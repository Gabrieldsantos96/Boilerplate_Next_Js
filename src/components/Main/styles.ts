'use client'

import styled, { css } from 'styled-components'
import { Wrapper as Toggle } from '@/components/Toggle/styles'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    background-color: ${theme.colors.background};
    width: 100%;
    height: 100%;
    padding: 1rem;
    ${Toggle} {
      flex-direction: row-reverse;
      margin-bottom: ${theme.spacings.xsmall};
    }
  `}
`
export const Content = styled.main`
  height: calc(100% - 4.1rem);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`

export const Title = styled.h1`
  font-size: 2.5rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`

export const Illustration = styled.img`
  margin-top: 3rem;
  width: min(30rem, 100%);
`
