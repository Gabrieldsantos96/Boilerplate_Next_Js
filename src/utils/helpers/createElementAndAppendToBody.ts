export function createElementAndAppendToBody(containerId: string) {
  const wrapperElement = document.createElement('div')
  wrapperElement.setAttribute('id', containerId)
  wrapperElement.setAttribute('data-testid', containerId)
  document.body.appendChild(wrapperElement)
  return wrapperElement
}
