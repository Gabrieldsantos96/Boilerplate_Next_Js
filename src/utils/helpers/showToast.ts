import { EventManagerDataTransfer } from '@/components/Toast/ToastContainer/useToastContainer'
import EventManager from '@/lib/eventManager'

export const toastEventManager = new EventManager<EventManagerDataTransfer>()

export const showToast = ({
  type,
  text,
  duration
}: EventManagerDataTransfer) => {
  toastEventManager.emit('addToast', { type, text, duration })
}
