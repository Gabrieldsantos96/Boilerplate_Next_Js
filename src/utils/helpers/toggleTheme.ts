import EventManager from '@/lib/eventManager'

export const themeEventManager = new EventManager<string>()

export const toggleTheme = () => {
  themeEventManager.emit('toggleTheme', '')
}
