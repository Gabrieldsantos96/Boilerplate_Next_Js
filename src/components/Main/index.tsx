'use client'

import { toggleTheme } from '@/utils/helpers/toggleTheme'
import * as S from './styles'
import Checkbox from '../Toggle'

const Main = ({
  title = 'Boilerplate Next JS',
  description = 'TypeScript, ReactJS, NextJS e Styled Components'
}) => {
  return (
    <S.Wrapper>
      <Checkbox onCheck={toggleTheme} label="Toggle theme" />
      <S.Content>
        <S.Title>{title}</S.Title>
        <S.Description>{description}</S.Description>
        <S.Illustration
          src="/img/hero-illustration.svg"
          alt="Um desenvolvedor de frente para uma tela com cÃ³digo."
        />
      </S.Content>
    </S.Wrapper>
  )
}

export default Main
