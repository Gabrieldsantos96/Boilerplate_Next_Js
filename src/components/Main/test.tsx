import { screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { render } from '@/utils/helpers/testUtils'
import Main from '.'

jest.mock('@/utils/helpers/toggleTheme', () => ({
  __esModule: true,
  toggleTheme: jest.fn()
}))

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)
    expect(container.firstChild).toHaveStyle({
      width: '100%',
      height: '100%',
      padding: '1rem'
    })
  })
})
