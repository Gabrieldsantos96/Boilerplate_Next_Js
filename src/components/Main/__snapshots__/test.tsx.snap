// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c2 {
  display: flex;
  align-items: center;
}

.c4 {
  margin-left: 0.8rem;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  appearance: none;
  width: 6rem;
  height: 2.5rem;
  background-color: #06092b;
  border-radius: 0.4rem;
  transition: background-color 0.3s ease-in-out;
  position: relative;
  outline: none;
}

.c4:before {
  content: '';
  height: 1em;
  width: 1em;
  border-radius: 1em;
  background-color: #F6F5FC;
  position: absolute;
  left: 0.2rem;
  transition: left 0.2s linear;
}

.c3 {
  cursor: pointer;
  padding-left: 0.8rem;
  color: #222222;
  line-height: 1.8rem;
}

.c0 {
  background-color: #F6F5FC;
  width: 100%;
  height: 100%;
  padding: 1rem;
}

.c0 .c1 {
  flex-direction: row-reverse;
  margin-bottom: 1.6rem;
}

.c5 {
  height: calc(100% - 4.1rem);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.c6 {
  font-size: 2.5rem;
}

.c7 {
  font-size: 2rem;
  font-weight: 400;
}

.c8 {
  margin-top: 3rem;
  width: min(30rem, 100%);
}

<main
  class="c0"
>
  <div
    class="c1 c2"
  >
    <label
      class="c3"
      for=""
    >
      Toggle theme
    </label>
    <input
      class="c4"
      id=""
      type="checkbox"
    />
  </div>
  <main
    class="c5"
  >
    <h1
      class="c6"
    >
      Boilerplate Next JS
    </h1>
    <h2
      class="c7"
    >
      TypeScript, ReactJS, NextJS e Styled Components
    </h2>
    <img
      alt="Um desenvolvedor de frente para uma tela com cÃ³digo."
      class="c8"
      src="/img/hero-illustration.svg"
    />
  </main>
</main>
`;
